climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+                (no reference)
+            return_plot : bool, optional, default=False
+                indicate if the output plot of the concave hull algorithm
+                should be returned
+
+        Returns
+        -------
+            burned area : float
+            plot : optional
+
+        See also
+        --------
+        climada.util.alpha_shape
+        https://towardsdatascience.com/the-concave-hull-c649795c0f0f
+        
+        """
+        ev_idx = np.argwhere(self.event_id == ev_id).reshape(-1)[0]
+        if not ev_idx.size:
+            raise ValueError(No event with id %d found', ev_id)
+
+        if not self.centroids.lat.size:
+            self.centroids.set_meta_to_lat_lon()
+
+        # Extract coordinates where intensity > 0 (for a given fire)
+        fire_lat = self.centroids.lat[self.intensity[ev_idx, :].nonzero()[1]]
+        fire_lon = self.centroids.lon[self.intensity[ev_idx, :].nonzero()[1]]
+
+        # Creation of the geodataframe
+        fire = gpd.GeoDataFrame({'geometry': gpd.points_from_xy(fire_lon,  fire_lat)}, crs=DEF_CRS)
+        points = fire.geometry.values
+
+        # Compute concave hull
+        concave_hull, _ = u_alpha.alpha_shape(points, alpha=alpha)
+
+        # Compute area concave hull in right projection
+        project = partial(
+            pyproj.transform,
+            pyproj.Proj(init=DEF_CRS['init']), # source coordinate system
+            pyproj.Proj(init='epsg:3310')) # destination coordinate system: albers california
+
+        concave_hull_m = transform(project, concave_hull)  # apply projection
+        area_hull_one_fire = concave_hull_m.area/10000
+
+        # Plot the polygone around the fire
+        if return_plot:
+            u_alpha.plot_polygon(concave_hull)
+            plt.plot(fire_lon, fire_lat, 'o', color='red', markersize=0.5)
+
+        return area_hull_one_fire
+
+    def set_hist_fire_seasons_FIRMS(self, csv_firms, centr_res_factor=1,
+                                    centroids=None, hemisphere=None,
+                                    year_start=None, year_end=None,
+                                    keep_all_fires=False):
+
+        """ Parse FIRMS data and generate historical fire seasons.
+        
+        Individual fires are created using temporal and spatial clustering
+        according to the 'set_hist_fire' method. fires are then summarized
+        using max intensity at each centroid for each year.
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+                (https://firms.modaps.eosdis.nasa.gov/download/)
+            centr_res_factor : int, optional, default=1
+                resolution factor with respect to the satellite data to use
+                for centroids creation
+            centroids : Centroids, optional
+                centroids in degrees to map data
```

Comment:
Regular grid etc. needs to be explained.

Reply:
adjusted - 82029e66